### PR TITLE
MAT-4943: Adding types for Obse3rvations and updating validators

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE PR Template

Jira Ticket: [MAT-4943](https://jira.cms.gov/browse/MAT-4943)
(Optional) Related Tickets:

### Summary

Added PopulationTypes that are used to distinguish between the measureObservations added to the measure populatinos and the generated observations added to test cases 

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e. Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/measure/PopulationType.java [hunk 1]
```diff
@@ -19,6 +19,12 @@ public enum PopulationType {
   DENOMINATOR_EXCEPTION("denominator-exception","Denominator Exception"),
   @JsonProperty("measurePopulation")
   MEASURE_POPULATION("measure-population","Measure Population"),
+  @JsonProperty("measurePopulationObservation")
+  MEASURE_POPULATION_OBSERVATION("measure-population-observation","Measure Population Observation"),
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/models/measure/PopulationType.java:23]: I think the code and display for all three measure observation types should just be "measure-observation" and "Measure Observation" cause these were tied to the FHIR types

File: src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java [hunk 2]
```diff
@@ -50,33 +51,55 @@ public boolean isValid(
...
-        try {
-          if(!popVal.getName().equals(PopulationType.MEASURE_OBSERVATION)){
-            Integer.parseInt(popVal.getExpected().toString());
-            return true;
+        boolean isObv = isObservation(popVal.getName());
+        boolean getExp = popVal.getExpected().equals(false);
+        //isObservation needs to be a number, regardless of populationBasis
+        //!isObservation needs to match popBasis type (ie., Boolean or Number)
+
+        if(isObservation(popVal.getName())) {
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java:63]: did you mean to use your `isObv` variable from a few lines up

File: src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java [hunk 3]
```diff
@@ -50,33 +51,55 @@ public boolean isValid(
...
+      
+      if (popVal == null || popVal.getExpected() == null || popVal.getExpected().toString().isEmpty()) {
         return true;
       } else {
-        try {
-          if(!popVal.getName().equals(PopulationType.MEASURE_OBSERVATION)){
-            Integer.parseInt(popVal.getExpected().toString());
-            return true;
+        boolean isObv = isObservation(popVal.getName());
+        boolean getExp = popVal.getExpected().equals(false);
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java:59]: are either of these vars used anywhere

File: src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java (hunk 2 above)
Comment (gregory-akins) [reply src/main/java/gov/cms/madie/models/validators/ScoringPopulationValidator.java:60]: Nope.. just forgot to clean that up; fixed now

File: src/main/java/gov/cms/madie/models/validators/GroupScoringPopulationValidator.java [hunk 4]
```diff
@@ -28,35 +29,53 @@ public boolean isValid(Group group, ConstraintValidatorContext context) {
...
-              .orElse(null);
+
+            Stream<Population> pops = populations.stream()
+                .filter(population -> {
+
+                  return Objects.equals(option.getMeasurePopulation(), population.getName()); 
+                });
+            
+            
+            Population matchingPopulation =  pops
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/models/validators/GroupScoringPopulationValidator.java:56]: curious why this single stream operation was split into two chunks

Comment (gregory-akins) [reply src/main/java/gov/cms/madie/models/validators/GroupScoringPopulationValidator.java:56]: To make it easier to debug

**Review by nmorasb — APPROVED** [review]

**Pull request merged**